sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,549 @@
+			fmt.Fprintln(v.stringBuilder)
+		}
+	}
+	fmt.Fprintln(v.stringBuilder, "@enduml")
+	return v.stringBuilder.String()
+
+}
+
+func (v *IntsDiagramVisitor) drawComponentView(viewParams viewParams, params *IntsParam, nameMap map[string]string) {
+	callsDrawn := map[CallsDrawn]struct{}{}
+	if viewParams.endptAttrs["view"].GetS() == "system" {
+		v.drawSystemView(viewParams, params, nameMap)
+	} else {
+		for _, dep := range params.integrations {
+			appA := dep.Self.Name
+			appB := dep.Target.Name
+			apps := &CallsDrawn{
+				Self:   appA,
+				Target: appB,
+			}
+			var direct []string
+			if _, ok := params.highlights[appA]; ok {
+				direct = append(direct, appA)
+			}
+			if _, ok := params.highlights[appB]; ok {
+				direct = append(direct, appB)
+			}
+			_, ok := callsDrawn[*apps]
+			if appA != appB && !ok {
+				if direct != nil || viewParams.indirectArrowColor != "none" {
+					indirect := ""
+					if direct == nil {
+						indirect = " <<indirect>>"
+					}
+					fmt.Fprintf(v.stringBuilder, "%s --> %s%s", v.VarManagerForComponent(appA, nameMap), v.VarManagerForComponent(appB, nameMap), indirect)
+					fmt.Fprintln(v.stringBuilder)
+					callsDrawn[*apps] = struct{}{}
+				}
+			}
+		}
+		for _, app := range params.apps {
+			for _, mixin := range v.mod.Apps[app].GetMixin2() {
+				mixinName := strings.Join(mixin.Name.Part, " :: ")
+				fmt.Fprintf(v.stringBuilder, "%s <|.. %s", v.VarManagerForComponent(mixinName, nameMap), v.VarManagerForComponent(app, nameMap))
+				fmt.Fprintln(v.stringBuilder)
+			}
+		}
+	}
+}
+
+func (v *IntsDiagramVisitor) drawSystemView(viewParams viewParams, params *IntsParam, nameMap map[string]string) {
+	callsDrawn := map[CallsDrawn]struct{}{}
+	for _, dep := range params.integrations {
+		appA := dep.Self.Name
+		appB := dep.Target.Name
+		apps := &CallsDrawn{
+			Self:   appA,
+			Target: appB,
+		}
+		var direct []string
+		if _, ok := params.highlights[appA]; ok {
+			direct = append(direct, appA)
+		}
+		if _, ok := params.highlights[appB]; ok {
+			direct = append(direct, appB)
+		}
+		appA = strings.Split(appA, " :: ")[0]
+		appB = strings.Split(appB, " :: ")[0]
+		_, ok := callsDrawn[*apps]
+		if appA != appB && !ok {
+			if direct != nil || viewParams.indirectArrowColor != "none" {
+				indirect := ""
+				if direct == nil {
+					indirect = " <<indirect>>"
+				}
+				fmt.Fprintf(v.stringBuilder, "%s --> %s%s", v.VarManagerForComponent(appA, nameMap), v.VarManagerForComponent(appB, nameMap), indirect)
+				fmt.Fprintln(v.stringBuilder)
+				callsDrawn[*apps] = struct{}{}
+			}
+		}
+	}
+}
+
+func (v *IntsDiagramVisitor) generateComponentView(args *Args, viewParams viewParams, params *IntsParam) string {
+
+	fmt.Fprintln(v.stringBuilder, "@startuml")
+	if viewParams.diagramTitle != "" {
+		fmt.Fprintln(v.stringBuilder, "title "+viewParams.diagramTitle)
+	}
+	fmt.Fprintln(v.stringBuilder, "hide stereotype")
```

Comment:
@JackLeeHal use golang multi-line string for constant strings. No need to make function calls.
And why not use v.StringBuilder.WriteString() method directly?

Reply:
use multi-line string and changed to v.StringBuilder.WriteString()